### DISCUSSION TITLE
How can I use my * certificate to host argocd ?

### DISCUSSION THREAD

**Discussion by philippe-lavoie** [argoproj/argo-cd#6204] [Q&A]
I've added argocd to my cluster, however connecting to it through a load balancer fails (invalid certificate). My cluster is managed by a wildcard certificate to mixing letsencrypt with it fails. Is there any documentation on how to achieve this ? I only see letsencrypt related information.

**Comment by jgwest** [comment]
Hi @philippe-lavoie , this would probably depend on your individual ingress/load balancer setup, rather than anything specific to Argo CD. You can see the [ingress configuration](https://argoproj.github.io/argo-cd/operator-manual/ingress/) page for description of the ports Argo CD listens on, and ideas on how to accomplish this with various ingresses/LBs. 

My only suggestion is you may want to consider this, if you've not already tried it: "The API server should be run with TLS disabled. Edit the argocd-server deployment to add the --insecure flag to the argocd-server command.", if you are doing TLS termination within the cluster.

If anyone has any specific suggestions, feel free to reply.

**Reply by philippe-lavoie** [reply]
Not sure why, but when I expose it, I get a 404 from argocd. I'll keep using port forwarding for now and keep on clicking yes when it says it's insecure.

**Reply by stieglma** [reply]
I have Argo CD running behind an nginx Ingress with cert-manager for lets-encrypt certificates. For that to work you have to specify some labels on the ingress:
```
nginx.ingress.kubernetes.io/ssl-passthrough: "true"
nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
```
This is also described here: https://argoproj.github.io/argo-cd/operator-manual/ingress/#option-1-ssl-passthrough contrary to this documentation I however didn't need to change the commandline arguments of the nginx-ingress-controller
